packages/web-api/src/file-upload.ts

```diff
@@ -47,10 +45,16 @@ export async function getFileUploadJob(
       ...fileUploadJob,
     };
   }
-  return {
-    file: options.file,
-    ...fileUploadJob,
-  };
+  if ('file' in options) {
+    return {
+      file: options.file,
+      ...fileUploadJob,
+    };
+  }
+  throw errorWithCode(
+    new Error('Either a file or content field is required for valid file upload. You must supply one'),
+    ErrorCode.FileUploadInvalidArgumentsError,
+  );
```

Comment:
Changes to check for the existence of `file` or whatever key is being done for stronger type safety. The error case would be thrown by the API and is moved ahead a bit here. I think this is also checked earlier in the call too, but isn't picked up by the types.